docs/DevEnv.md

```diff
@@ -0,0 +1,66 @@
+## Netplugin Developer's Guide
+This document describes the development environment and typical steps to hack on netplugin.
+While there are alternative ways, the steps listed here is used by a lot of developers while
+making incremental changes and doing frequent compilation for a quicker development cycle.
+
+Happy hacking!
+
+### 1. Check-out a tree. `Estimated time: 2 minutes`
+Notes: Make sure GOPATH is set. This is a one time activity.
```

Comment:
@jainvipin 

> The reason I prefer setting GOPATH is because it allows for entire GOPATH to be available inside instead of just netplugin.

I just looked at Vagrantfile (https://github.com/contiv/netplugin/blob/master/Vagrantfile#L122), it doesn't mounts the entire host's GOPATH but just the netplugin repo. But the Vagrantfile still depends on GOPATH being defined in host's environment, which is what I was hoping that we could skip.

@unclejack 

> Is there a reason behind not wanting to bind mount this into the VM's GOPATH?

As I discussed above the netplugin repo is indeed being bind mounted at GOPATH in side the vm. My concern is more around the work needed for having GOPATH to be set in the host itself.